### PR TITLE
Synchronize calls to getenv

### PR DESCRIPTION
**Pull request by dominiklohmann**
###  :notebook_with_decorative_cover: Description

As indicated by clang-tidy, getenv and friends are not thread-safe. This adds wrapper functions that synchronize the calls to them.

###  :memo: Checklist

- [x] All user-facing changes have changelog entries.
- [x] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.

### :dart: Review Instructions

File-by-file.

### CODE REVIEW

File: libvast/vast/detail/env.hpp [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+#include <string>
+
+namespace vast::detail {
+
+/// A thread-safe wrapper around `::getenv`.
+/// @param var The environment variable.
+/// @returns The environment variables contents, or an empty string if the
+/// variable is unset.
+std::string env(const char* var);
```
Comment (mavam) [inline libvast/vast/detail/env.hpp:19]: Returning `std::optional<std::string>` would make it possible to distinguish between an unset and a set variable that is empty.

Comment (dominiklohmann) [reply libvast/vast/detail/env.hpp]: Done.

File: libvast/vast/detail/env.hpp [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+/// A thread-safe wrapper around `::getenv`.
+/// @param var The environment variable.
+/// @returns The copied environment variables contents, or `std::nullopt`.
+[[nodiscard]] std::optional<std::string> env(const char* var);
+
+/// A thread-safe wrapper around `::unsetenv`.
+/// @param var The environment variable.
+/// @returns True on success.
+[[nodiscard]] bool unset_env(const char* var);
```
Comment (lava) [inline libvast/vast/detail/env.hpp:24]: Nodiscard seems a bit overzealous here, or if you want to force error handling you should at least return a `caf::error` and/or the proper errno.

File: libvast/vast/detail/env.hpp [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+#include <optional>
+#include <string>
+
+namespace vast::detail {
+
+/// A thread-safe wrapper around `::getenv`.
+/// @param var The environment variable.
+/// @returns The copied environment variables contents, or `std::nullopt`.
+[[nodiscard]] std::optional<std::string> env(const char* var);
```
Comment (lava) [inline libvast/vast/detail/env.hpp:19]: I'd propose we stay close to the wrapped functions and call these `getenv/unsetenv`? Or maybe even something like `getenv_mt`/`locked_getenv`/`blocking_getenv`/`threadsafe_getenv` to make it clear at the callsite why we don't just use the standard library functions.

File: libvast/src/detail/env.cpp [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+#include "vast/detail/env.hpp"
+
+#include <cstdlib>
+#include <mutex>
+
+namespace vast::detail {
+
+namespace {
+
+auto env_mutex = std::mutex{};
```
Comment (lava) [inline libvast/src/detail/env.cpp:18]: What do you think in general about addind a `s_` prefix for the name of static data? For these small functions its obvious, but in general i think its nice when access to global data really stands out while reading the source code. In particular once classes get thrown into the mix, where a non-locally declared variable might be either a data member or a global variable.

**Review by lava — APPROVED** [review]
Looks good to me except for some bikeshedding below.

File: libvast/vast/detail/env.hpp (hunk 3 above)
Comment (dominiklohmann) [reply libvast/vast/detail/env.hpp:19]: I like the `locked_` prefix, will change.

File: libvast/src/detail/env.cpp (hunk 4 above)
Comment (dominiklohmann) [reply libvast/src/detail/env.cpp:18]: I will add a comment above the variable that makes it clear that this is global to this translation unit.

File: libvast/vast/detail/env.hpp (hunk 2 above)
Comment (dominiklohmann) [reply libvast/vast/detail/env.hpp:24]: I like the latter. Will do.

**Pull request merged**